src/modules/dataman/dataman.cpp

```diff
@@ -540,7 +540,15 @@ _file_write(dm_item_t item, unsigned index, dm_persitence_t persistence, const v
 static void
 _ram_flash_update_flush_timeout()
 {
-	dm_operations_data.ram_flash.flush_timeout_usec = hrt_absolute_time() + RAM_FLASH_FLUSH_TIMEOUT_USEC;
+	timespec &abstime = dm_operations_data.ram_flash.flush_timeout;
+
+	if (clock_gettime(CLOCK_REALTIME, &abstime) == 0) {
```

Comment:
I think you should use `px4_clock_gettime`.

Reply:
We can, but see above about platform-specifics.